crates/iceberg/src/spec/schema.rs

```diff
@@ -86,6 +87,16 @@ impl SchemaBuilder {
         self
     }
 
+    /// Reassign all field-ids (nested) on build.
+    /// If `start_from` is provided, it will start reassigning from that id (inclusive).
+    /// If not provided, it will start from 0.
+    ///
+    /// All specified aliases and identifier fields will be updated to the new field-ids.
+    pub fn with_reassigned_field_ids(mut self, start_from: Option<i32>) -> Self {
+        self.reassign_field_ids_from = Some(start_from.unwrap_or(0));
```

Comment:
If we always set `Some(v)` here, how about just using `i32` for `reassign_field_ids_from` with default value `0`?

Reply:
Typically you would pass in `last-column-id` from the table metadata. In the case of a `CREATE TABLE` this will be zero, in the case of a `REPLACE TABLE` it will be `last-column-id` where all columns get a new ID and you ensure they are not used before.

Reply:
@Xuanwo only doubt with the i32 is that people might input 1 as something is required - and it's shorter than Default::default().
The rust tests are quite a good example that this can happen: We have many tests where field numbering starts from 1, so I had to re-write them to be compatible with the new builder.

I am OK with the i32 as well, just wanted to bring up that a None is probably more verbose.

Reply:
Both good points, fixed in the latest commit.
Its `dead_code` for now